src/vstest.console/Internal/ConsoleLogger.cs

```diff
@@ -355,53 +354,82 @@ private void TestResultHandler(object sender, TestResultEventArgs e)
-                        name);
-                    Output.Information(false, ConsoleColor.Red, output);
-                    DisplayFullInformation(e.Result);
-                }
-            }
-            else if (e.Result.Outcome == TestOutcome.Passed)
+            string testDisplayName = e.Result.DisplayName;
+            if (string.IsNullOrWhiteSpace(e.Result.DisplayName))
             {
-                if (this.verbosityLevel.Equals(Verbosity.Normal))
-                {
-                    var output = string.Format(CultureInfo.CurrentCulture, CommandLineResources.PassedTestIndicator, name);
-                    Output.Information(false, output);
-                    DisplayFullInformation(e.Result);
-                }
-                this.testsPassed++;
+                testDisplayName = e.Result.TestCase.DisplayName;
             }
-            else
+
+            switch (e.Result.Outcome)
             {
-                if (!this.verbosityLevel.Equals(Verbosity.Quiet))
-                {
-                    var output = string.Format(CultureInfo.CurrentCulture, CommandLineResources.NotRunTestIndicator,
-                        name);
-                    Output.Information(false, output);
-                    DisplayFullInformation(e.Result);
-                }
+                case TestOutcome.Skipped:
+                    {
+                        testsSkipped++;
+                        if (verbosityLevel == Verbosity.Quiet)
+                        {
+                            break;
+                        }
+
+                        var output = string.Format(CultureInfo.CurrentCulture, CommandLineResources.SkippedTestIndicator, testDisplayName);
+                        Output.Warning(false, output);
+                        if (verbosityLevel == Verbosity.Detailed)
+                        {
+                            DisplayFullInformation(e.Result);
+                        }
+
+                        break;
+                    }
+
+                case TestOutcome.Failed:
+                    {
+                        testsFailed++;
+                        if (verbosityLevel == Verbosity.Quiet)
+                        {
+                            break;
+                        }
+
+                        var output = string.Format(CultureInfo.CurrentCulture, CommandLineResources.FailedTestIndicator, testDisplayName);
+                        Output.Information(false, ConsoleColor.Red, output);
+                        DisplayFullInformation(e.Result);
+                        break;
+                    }
+
+                case TestOutcome.Passed:
+                    {
+                        testsPassed++;
+                        if (verbosityLevel == Verbosity.Quiet)
+                        {
+                            break;
+                        }
+
+                        var output = string.Format(CultureInfo.CurrentCulture, CommandLineResources.PassedTestIndicator, testDisplayName);
```

Comment:
On Minimal level the passed test name should not get displayed. Update the test to catch this scenario.

Reply:
Fixed